dashboard/app/views/devise/registrations/edit.html.haml

```diff
@@ -5,6 +5,8 @@
 %h2= @page_title
 
 = form_for(resource, :as => resource_name, :url => registration_path(resource_name), :html => { :method => :put }) do |f|
+  - email_mismatch = current_user.errors.delete(:email_mismatch)
+  - current_user.reload if email_mismatch.present?
```

Comment:
Part of the spec for this feature is to revert the form to its previous state if the email does not match, rather than retaining the attempted changes, so we deliberately reload here if that particular error is encountered (the errors remain intact).